Makefile

```diff
@@ -19,50 +26,48 @@ $(error "error Unexpected OS; Only Linux or Darwin supported.")
 endif
 endif
 
-# Environment
-SHELL := /bin/bash
-BASE_PATH := $(shell pwd)
-PUBLIC_PATH := $(BASE_PATH)/public
-THEMES_PATH := $(BASE_PATH)/themes
-STATIC_PATH := $(BASE_PATH)/static
-THEME_NAME := $(shell basename $(HUGO_THEME))
-THEME_PATH := $(THEMES_PATH)/$(THEME_NAME)
-HUGO_PATH := $(BASE_PATH)/.hugo
-HUGO_URL = https://github.com/spf13/hugo/releases/download/v$(HUGO_VERSION)/$(HUGO_TAR_FILE_NAME)
-HUGO_NAME := hugo_$(HUGO_VERSION)_$(ARCH)_$(OS)
-HUGO_URL_NAME := hugo_$(HUGO_VERSION)_$(URL_ARCH)-$(URL_OS)
-
 # Tools
-CURL = curl -L
-HUGO = $(HUGO_PATH)/hugo
-MKDIR = mkdir -p
-GIT = git
+CURL := curl -L
+HUGO := $(HUGO_PATH)/hugo
+MKDIR := mkdir -p
+JS_PACKAGE_MANAGER := yarn
 UNCOMPRESS := tar -zxf
 
 # Rules
 all: build
 
-init:
-	@if [ "$(HUGO_THEME)" == "" ]; then \
-		echo "ERROR! Please set the env variable 'HUGO_THEME' (http://mcuadros.github.io/autohugo/documentation/working-with-autohugo/)"; \
-		exit 1; \
-	fi;
+# Ensures the Hugo binary existance
+$(HUGO): $(HUGO_TAR_PATH)
+	$(UNCOMPRESS) $(HUGO_TAR_PATH) --directory=$(HUGO_PATH);
+
+# Downloads the hugo binary
+$(HUGO_TAR_PATH):
+	echo "Downloading $(HUGO_URL)"
+	$(MKDIR) $(HUGO_PATH)
+	$(CURL) $(HUGO_URL) -o $(HUGO_TAR_PATH)
 
-dependencies: init
-	@if [[ ! -f $(HUGO) ]]; then \
-		$(MKDIR) $(HUGO_PATH); \
-		cd $(HUGO_PATH); \
-		echo "Downloading $(HUGO_URL)"; \
-		$(CURL) $(HUGO_URL) -o $(HUGO_TAR_FILE_NAME); \
-		$(UNCOMPRESS) $(HUGO_TAR_FILE_NAME); \
-	fi;
+# Ensures hugo dependencies
+hugo-dependencies: $(HUGO_TAR_PATH) $(HUGO)
```

Comment:
why have you removed the condition? Why have you separated this in two steps that are only called from here?

Reply:
I tried to be more modular for the two different cases: controlling the proper tar version, downloading if it is not available the proper one and performing the uncompress. With that, it is even avoided the `if`s and the commands are outputted to the log, what is more clear while developing and debugging

Reply:
Will this be run if the correct version is already there, downloaded and uncompressed?

Reply:
nope. If the corrent version is already available, that rules does not run